### PR TITLE
validation: add env validation for hooks

### PR DESCRIPTION
**Pull request by Mashimiao**
Signed-off-by: Ma Shimiao <mashimiao.fnst@cn.fujitsu.com>

### CODE REVIEW

File: validate.go [hunk 1]
```diff
@@ -125,12 +125,12 @@ func checkPlatform(platform rspec.Platform) {
...
-	checkEventHookPaths("pre-start", hooks.Prestart, hooksCheck)
-	checkEventHookPaths("post-start", hooks.Poststart, hooksCheck)
-	checkEventHookPaths("post-stop", hooks.Poststop, hooksCheck)
+	hookValid("pre-start", hooks.Prestart, hooksCheck)
+	hookValid("post-start", hooks.Poststart, hooksCheck)
+	hookValid("post-stop", hooks.Poststop, hooksCheck)
 }
 
-func checkEventHookPaths(hookType string, hooks []rspec.Hook, hooksCheck bool) {
+func hookValid(hookType string, hooks []rspec.Hook, hooksCheck bool) {
```
Comment (wking) [inline validate.go]: `checkEventHooks`?

File: validate.go [hunk 2]
```diff
@@ -144,6 +144,11 @@ func checkEventHookPaths(hookType string, hooks []rspec.Hook, hooksCheck bool) {
 			if fi.Mode()&0111 == 0 {
 				logrus.Fatalf("The %s hook %v: is not executable", hookType, hook.Path)
 			}
+			for _, env := range hook.Env {
+				if !envValid(env) {
+					logrus.Fatalf("Env %q for hook %v is in the invalid form.", env, hook.Path)
+				}
+			}
```
Comment (wking) [inline validate.go]: This check is host-independent, so it should happen regardless of whether `--hooks` is set (and that portability was the main reason for the flag in #49.  For example, see [here][1]).

[1]: https://github.com/opencontainers/ocitools/pull/49#issuecomment-216585596

**Comment by wking** [comment]
Two minor nits (I left inline comments), but otherwise 62ab8b7f looks
reasonable to me.

File: validate.go (hunk 1 above)
Comment (Mashimiao) [inline validate.go]: checkEventHook?

Comment (Mashimiao) [inline validate.go]: I think `checkEventHooks` is OK.

File: validate.go (hunk 2 above)
Comment (Mashimiao) [inline validate.go]: fixed, thanks.

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**